### PR TITLE
Add page specific IDs

### PR DESCRIPTION
**Pull request by hgoncalves**
Add an id for each page canvas.

This makes it easier to detect the page on which an electronic signature has been dragged and dropped, for example, before actually digitally signing the document.

### CODE REVIEW

**Comment by hgoncalves** [comment]
Hi @hrynko, thanks just added the description and a use case

**Comment by hrynko** [comment]
I find it not quite flexible. Can you consider adding an `id` prop for the component, which, if defined, would naturally be assigned to the wrapper `div` element, and beyond that to all child `svg` elements, but postfixed with page numbers?

So that `<vue-pdf-embed id="my-pdf" ... />` will result in:

```html
<div id="my-pdf">
  <svg id="my-pdf-1" ... />
  <svg id="my-pdf-2" ... />
  <svg id="my-pdf-3" ... />
</div>
```

**Comment by hgoncalves** [comment]
Hi, you're absolutely right I will do that and let you know as soon as it's done.

**Comment by hgoncalves** [comment]
Just pushed the changes you suggested

File: src/vue-pdf-embed.vue [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
-    <canvas v-for="pageNum in pageNums" :key="pageNum"></canvas>
+  <div v-if="documentId" :id="documentId">
+    <canvas
+      v-for="pageNum in pageNums"
+      :key="pageNum"
+      :id="`${documentId}-${pageNum}`"
+    />
+  </div>
+  <div v-else>
+    <canvas v-for="pageNum in pageNums" :key="pageNum" />
```
Comment (hrynko) [inline src/vue-pdf-embed.vue:10]: Let's just override `id` for simplicity

```suggestion
  <div :id="id">
    <canvas
      v-for="pageNum in pageNums"
      :key="pageNum"
      :id="id && `${id}-${pageNum}`"
    />
```

File: src/vue-pdf-embed.vue [hunk 2]
```diff
@@ -13,6 +20,14 @@ pdf.GlobalWorkerOptions.workerPort = new PdfWorker()
 export default {
   name: 'VuePdfEmbed',
   props: {
+    /**
+     * Id of the document to display 
+     * @values String
+     */
+    documentId: {
```
Comment (hrynko) [inline src/vue-pdf-embed.vue:27]: In addition to the above

```suggestion
    id: {
```

File: src/vue-pdf-embed.vue [hunk 3]
```diff
@@ -13,6 +20,14 @@ pdf.GlobalWorkerOptions.workerPort = new PdfWorker()
...
+      type: String,
+      required: false,
```
Comment (hrynko) [inline src/vue-pdf-embed.vue:29]: Default value should be set for each props that are not marked as `required` (https://eslint.vuejs.org/rules/require-default-prop.html)

```suggestion
      default: null,
```

File: src/vue-pdf-embed.vue [hunk 4]
```diff
@@ -13,6 +20,14 @@ pdf.GlobalWorkerOptions.workerPort = new PdfWorker()
 export default {
   name: 'VuePdfEmbed',
   props: {
+    /**
+     * Id of the document to display
```
Comment (hrynko) [inline src/vue-pdf-embed.vue:24]: Technically, this is not a document ID

```suggestion
     * Component identifier (inherited by child SVGs with page number
     * postfixes).
```

**Review by hrynko — CHANGES_REQUESTED** [review]
Please also revert `package-lock.json` changes, since this PR does not change dependencies.

**Comment by hgoncalves** [comment]
Done ;)

**Review by hrynko — APPROVED** [review]

**Review by hrynko — CHANGES_REQUESTED** [review]
I don't think `pageClass` prop is necessary, since classes are flexible enough to be added as follows:

```html
<template>
  <vue-pdf-embed class="my-pdf" ... />
</template>

<style>
.my-pdf svg {
  ...
}
</style>
```

Please revert 9261bda

**Review by hrynko — APPROVED** [review]

**Comment by hrynko** [comment]
Thank you for your contribution @hgoncalves!

These changes will be included in the next release.

**Comment by hgoncalves** [comment]
> Thank you for your contribution @hgoncalves!
> 
> These changes will be included in the next release.

Thanks to you, for this package and your disposal.

**Comment by hrynko** [comment]
UPD: These changes were included in `1.0.8`

**Comment by hgoncalves** [comment]
Nice thanks

**Pull request merged**